tools/m2ctx/m2ctx.sh

```diff
@@ -0,0 +1,48 @@
+#!/usr/bin/env bash
+
+TMP_IMPORTS=$(pwd)/tmp-imports.c
+TMP_DECLARES=$(pwd)/tmp-declares.c
+OUT_FILE=ctx.c
+
+GCC=gcc
+FLAGS="-E -P -dD -undef"
+INCLUDES="-Iinclude -Iinclude/library -Ifiles -Ilib/include -include global.h"
+DEFINES="-DHEARTGOLD -DGAME_REMASTER=0 -DENGLISH -DPM_KEEP_ASSERTS -DSDK_ARM9 -DSDK_CODE_ARM -DSDK_FINALROM"
+
+
+generate-ctx () {
+    grep "^#include " "$1" > $TMP_IMPORTS
+    $GCC $FLAGS $INCLUDES $DEFINES $TMP_IMPORTS > $TMP_DECLARES
+
+    # Remove any line containing a predefined macro. If not removed, mwccarm
+    # generates compiler warnings.
+    sed '/__STDC__\|__STDC_VERSION__\|__STDC_VERSION__\|__STDC_HOSTED__/d' $TMP_DECLARES > $OUT_FILE
```

Comment:
No need to use temp files, just pipe

```suggestion
    grep "^#include " "$1" | $GCC $FLAGS $INCLUDES $DEFINES $TMP_IMPORTS -x c - | sed '/__STDC__\|__STDC_VERSION__\|__STDC_VERSION__\|__STDC_HOSTED__/d' > $OUT_FILE
```